bin/growpart

```diff
@@ -384,6 +395,52 @@ resize_sfdisk() {
 		max_end=$((${sector_num}-${gpt_second_size}))
 	fi
 
+	if [ -n "${free_percent}" ]; then
+		free_percent_sectors=$((sector_num/100*free_percent))
```

Comment:
Here we "floor" the number of free sectors; `$((sector_num*free_percent/100))` would be more precise, but I think what you did is fine, maybe even safer, even if we should still be afar from the bash arithmetic int overflow (64 bit signed).

One thing I was slightly worried about is partition alignment (see *align* in fdisk(8) and sfdisk(8)). At the moment, even without overprovisioning, growpart doesn't care about alignment, and just grows the partition to the last sector. This should be fine as aligning the end of the last partition is probably not critical, given that there are no more blocks the bad alignment can affect.

Should we care/worry about this when overprovisioning? Maybe we don't need to, trusting the SSD/SD firmware to do the right thing. Even if we do, then we should probably tackle the issue later in the script, for example by converting the amount to grow the partition by into a size in MiB, as sfdisk(8) recommends. I don't think think this has to be tackled in this PR.

This is more of a question than a review comment. @dermotbradley do you think above make sense, and do we agree it's not critical for landing this PR?

Reply:
to my knowledge, the *end* isn't really important for alignment except when it indicates (indirectly) the start of the next partition. when growpart grows to the full disk, there is no real value in end alignment and it can't easily affect beginning alignment. 

I guess if i were going to do things i'd go ahead and always end on a 1MiB-1 (or 4MiB-1) size but it doesn't seem terribly important.

Reply:
@paride I could change the order of the multiply/divide - indeed shellcheck recommends it the way you mentioned. Whichever way you prefer...

As for alignment, as @smoser pointed out growpart is only changing the end of the partition, not the start - in the majority of cases I'd assume its growing only the last (or only) partition on a device - that's certainly how I'm laying ouy my disk images, with rootfs as last partition. If its growing a partition other than the last one its still governed/restricted by the start of the next partition. So I don't believe alignment in this case is anything to worry about.

Reply:
Re-thinking about integer overflows, I think it's safer to first divide and then multiply to avoid breaking 32 bit systems.